tests/drivers/spi/spi_slave_master_loopback/src/spi.c

```diff
@@ -0,0 +1,425 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_LEVEL SYS_LOG_LEVEL_DEBUG
+#include <logging/sys_log.h>
+
+#include <zephyr.h>
+#include <misc/printk.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <spi.h>
+
+#define SPI_MASTER_CS (&spi_master_cs)
+#define SPI_SLAVE_CS (&spi_slave_cs)
+
+/* As configured in the board/.prj fragment */
+#define SPI_MASTER_DRV_NAME "SPI_LOOPBACK_MASTER"
+#define SPI_SLAVE_DRV_NAME "SPI_LOOPBACK_SLAVE"
+
+#if defined(CONFIG_BOARD_NUCLEO_F091RC)
+#define SPI_SLAVE 0
+#define MIN_FREQ 500000
+
+#undef SPI_MASTER_CS
```

Comment:
It looks like SPI_MASTER_CS ends up defined as null and MASTER_CS_CTRL_GPIO_DRV_NAME is "" no matter which branch is taken -- is that what you meant?

Reply:
I'm still not sure I know what your idea is here, since the definitions seem to end up the same.

Reply:
I left it open for new implementations...

Reply:
I simplified it with the Kconfig